crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -75,13 +75,19 @@ def current_step_state
       progress[:steps][current_step] || {}
     end
 
-    def start_step
+    # 'step' is name of the step user wants to start.
+    # If `out_of_order` is true, the step could be executed out of order
+    def start_step(step_name, out_of_order = false)
       ::Crowbar::Lock::LocalBlocking.with_lock(shared: false, logger: @logger, path: lock_path) do
-        if running?
+        if running? step_name
           @logger.warn("The step has already been started")
           return false
         end
-        progress[:steps][current_step][:status] = :running
+        if step_name != current_step && !out_of_order
```

Comment:
As mentioned in my comment below, those "out of order" steps are not always going to be "current" when running. They should be callable any time, even when some other step is "current".